### PR TITLE
Update deepcast extension

### PR DESCRIPTION
**Pull request by MonsterDeveloper**
## Description

1. Removed a redundant "Pro API" checkbox. According to [DeepL API Documentation](https://www.deepl.com/docs-api/api-access/authentication/), free API keys can be identified by `:fx` at the end.
2. Added support for switching languages when source is set to `Detect`. The value of detected source language is saved automatically and is set to target when switching languages.
3. Improved error handling.

#### Before
Switch languages — before
#### After
Switch languages — after

#### Before
Errors — before
#### After
Errors — after

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by pernielsentikaer** [comment]
@MonsterDeveloper Seems like there are some merge conflicts, could you look into this?

**Comment by pernielsentikaer** [comment]
@mooxl do you mind checking this?

File: extensions/deepcast/src/utils.ts [hunk 1]
```diff
@@ -1,34 +1,58 @@
...
 
 export function getPreferences() {
   return getPreferenceValues<Preferences>();
 }
 
+function isPro(key: string) {
+  return !key.endsWith(":fx");
+}
+
+function gotErrorToString(error: unknown) {
```
Comment (mooxl) [inline extensions/deepcast/src/utils.ts:17]: 🙏

**Review by mooxl — CHANGES_REQUESTED** [review]
Just a little nit:)

File: extensions/deepcast/src/utils.ts [hunk 2]
```diff
@@ -1,34 +1,58 @@
...
+    if (statusCode === StatusCodes.TOO_MANY_REQUESTS) return "Too many requests to DeepL API";
+    if (statusCode === 456)
+      return "DeepL API quota exceeded. The translation limit of your account has been reached. Consider upgrading your subscription.";
+    if (statusCode.toString().startsWith("5")) return "DeepL API is temporary unavailable. Please try again later.";
+
+    return `DeepL API returned ${statusCode} ${getReasonPhrase(statusCode)}`;
+  }
+
+  // request failed
+  if (error instanceof got.RequestError) return `Failed to send a request to DeepL API: ${error.code} ${error.message}`;
```
Comment (mooxl) [inline extensions/deepcast/src/utils.ts:31]: Could you please rephrase this message to `Something went wrong when sending a request to the DeepL API. If you’re having issues, open an issue on GitHub and include following text: ${error.code} ${error.message}`

**Review by mooxl — CHANGES_REQUESTED** [review]

File: extensions/deepcast/src/utils.ts (hunk 2 above)
Comment (MonsterDeveloper) [reply extensions/deepcast/src/utils.ts]: All seems resolved in the latest commit :)

**Review by mooxl — APPROVED** [review]
🚢

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**